suds/reader.py

```diff
@@ -58,7 +58,12 @@ def mangle(self, name, x):
         @rtype: str
 
         """
-        h = md5(name.encode()).hexdigest()
+        try:
+            h = md5(name.encode()).hexdigest()
+        except ValueError:
```

Comment:
Thanks - I'm wondering if this could have "false positives" of catching `ValueError`s (and then trying the version with `usedforsecurity`) when it shouldn't.

A few other options from looking at how others approach this: 
* Flip the order and catch AttributeError, e.g.
```
        try:
            # FIPS requires usedforsecurity=False and might not be
            # available on all distros: https://bugs.python.org/issue9216
            h = md5(name.encode(), usedforsecurity=False).hexdigest()
        except AttributeError:
            h = md5(name.encode()).hexdigest()
```
* A compatibility method: https://github.com/git-cola/git-cola/commit/5ae40d21279b73c1b35ce4f84c6abb075c50d3e2#diff-a5804055429122cbb41a6b906aa928641e4d7fee27c79dbe7bb47c726eeaeb73R32
* Just checking for python version >= 3.9 https://github.com/pytest-dev/pytest-randomly/pull/415/files#diff-77e4b024fe5132833f7d62b9ffc8028a856311e1a3afbb81f5715efc2bea28b9R5

If this is available in earlier versions of python on RHEL, I can see how checking for python 3.9 doesn't make sense.

No strong opinions on this, but let me know what you think.

Reply:
That sounds like a better way to do it. I've pushed the update.